src/epics/fetchAssetsEpic.js

```diff
@@ -240,3 +241,63 @@ export const fetchAssetsFromSharedListEpic = (action$, store, { client }) =>
           .catch(handleError)
       })
     })
+
+export const fetchAssetsFromSharedListWithFunctionEpic = (
+  action$,
+  store,
+  { client }
+) =>
+  action$
+    .ofType(actions.ASSETS_FETCH)
+    .filter(({ payload }) => {
+      return payload.type === 'list#shared' && payload.list.function
+    })
+    .mergeMap(({ payload }) => {
+      return Observable.from(
+        client.query({
+          query: projectsByFunctionGQL,
+          variables: { function: payload.list.function }
+        })
+      ).concatMap(({ data = {} }) => {
+        const startTime = Date.now()
+        const { allProjectsByFunction = [] } = data
+        const queries = allProjectsByFunction
+          .filter(({ slug }) => !!slug)
+          .map(({ slug }) => {
+            return client.query({
+              query: projectBySlugGQL,
+              variables: { slug },
+              context: { isRetriable: true }
+            })
+          })
+
+        if (allProjectsByFunction.length === 0) {
+          return Observable.of({
+            type: actions.ASSETS_FETCH_SUCCESS,
+            payload: {
+              items: [],
+              isLoading: false,
+              error: false
+            }
+          })
+        }
+
+        return Observable.forkJoin(queries)
+          .delayWhen(() => Observable.timer(500 + startTime - Date.now()))
```

Comment:
Is an arbitrary delay so necessary? Because it will cause a blink even when the data was cached.